api/app/signals/apps/questionnaires/fieldtypes/selected_object.py

```diff
@@ -0,0 +1,172 @@
+
+
+class SelectedObject(FieldType):
+    verbose_name = 'Selected object'
+    submission_schema = {
+        'type': 'object',
+        'properties': {
+            'id': {
+                'anyOf': [
+                    {
+                        'type': 'integer',
+                        'minimum': 0
+                    },
+                    {
+                        'type': 'string'
+                    }
+                ]
+            },
+            'label': {
+              'type': 'string'
+            },
+            'answer': {
+                'anyOf': [
+                    {
+                        'type': 'object',
+                        'required': [
+                            'type',
+                        ],
+                        'properties': {
+                            'type': {
+                                'type': 'string',
+                                'pattern': '^not-on-map$',
+                            },
+                        },
+                    },
+                    {
+                        'type': 'object',
+                        'required': [
+                            'id',
+                            'type',
+                            'label',
+                        ],
+                        'properties': {
+                            'id': {
+                                'type': 'integer',
+                                'minimum': 0,
+                            },
+                            'type': {
+                                'type': 'string',
+                                'pattern': '^not-on-map$',
+                            },
+                            'label': {
+                                'type': 'string',
+                            }
+                        },
+                    },
+                    {
+                        'type': 'object',
+                        'required': [
+                            'id',
+                            'type',
+                            'description',
+                            'label',
+                        ],
+                        'properties': {
+                            'id': {
+                                'anyOf': [
+                                    {
+                                        'type': 'integer',
+                                        'minimum': 0
+                                    },
+                                    {
+                                        'type': 'string'
+                                    }
+                                ]
+                            },
+                            'type': {
+                                'type': 'string',
+                            },
+                            'description': {
+                                'type': 'string',
+                            },
+                            'isReported': {
+                                'type': 'boolean',
```

Comment:
Is `isReported` general to several nuisance complaints flow? If it is not we may not want this here, same for the `location` property that takes a full address.

Reply:
Started with the FE implementation of storing a selected object in the extra_properties. This is something we need to discuss and agree upon what is still relevant.